### DISCUSSION TITLE
How to preview the built HTML Documentation in gh-pages branch?

### DISCUSSION THREAD

**Discussion by hwhsu1231** [chocolatey/docs#810] [Q&A] [answered]
Excuse me, Chocolatey Team.

Suppose that I only clone the `gh-pages` branch of [chocolatey/docs](https://github.com/chocolatey/docs). 

```cmd
git clone https://github.com/chocolatey/docs.git --branch gh-pages
```

How do I preview the built HTML documentation in this branch?

```cmd
D:\Repo\GitHub\testing\chocolatey-docs>git clone https://github.com/chocolatey/docs.git --branch gh-pages
Cloning into 'docs'...
remote: Enumerating objects: 285987, done.
remote: Counting objects: 100% (61353/61353), done.
remote: Compressing objects: 100% (7666/7666), done.
remote: Total 285987 (delta 53788), reused 61149 (delta 53625), pack-reused 224634
Receiving objects: 100% (285987/285987), 630.19 MiB | 3.68 MiB/s, done.
Resolving deltas: 100% (250377/250377), done.
Updating files: 100% (1205/1205), done.

D:\Repo\GitHub\testing\chocolatey-docs>chdir docs

D:\Repo\GitHub\testing\chocolatey-docs\docs>git branch --list
* gh-pages
```

**Comment by gep13** [comment]
@hwhsu1231 I guess there are a number of ways that this could be done, but I am wondering what the use case is here?  What is it you are actually trying to achieve?

One way that you could do it, assuming that you are using VSCode, would be to install this extension: https://marketplace.visualstudio.com/items?itemName=ritwickdey.LiveServer which allows you to serve the files from the current workspace that you have open in VSCode.

Outside of VSCode, and assuming that you are using Windows, would be to use IIS, to create a website pointed at the cloned directory.

**Comment by ryanrichter94** [comment, marked as answer]
Hi @hwhsu1231, our best practice is to use Docker to run a local dev container to view any changes you make in the Chocolatey docs repository. Our principal software engineer, Gary Ewan Park, has a YouTube video going over how to use this concept with Docker Desktop [here](https://youtu.be/HIHhxLBdVYw?si=XeKh8yeI-BlZ9XeA).
